src/Main.hs

```diff
@@ -1,16 +1,46 @@
-main = putStrLn "It works!"
+import Options.Applicative
 
-wrap :: Int -> String -> [String]
-wrap n s = foldl (splitWordsAt n) [] $ words s
+data Config = Config
+  { width :: Int
+  , path :: FilePath
+  }
+
+config :: Parser Config
+config = Config
+    <$> option auto
+        ( long "width"
+        <> short 'w'
+        <> metavar "WIDTH"
+        <> value 80
+        <> help "Specify a line width to use instead of the default 80 columns"
+        )
+
+    <*> argument str
+        ( metavar "PATH"
+        <> help "Path to input file"
+        )
+
+main :: IO ()
+main = execParser opts >>= run
+  where
+    opts = info (helper <*> config)
+      ( fullDesc
+      <> progDesc "Intelligently wrap lines to a given length"
+      )
+
+run :: Config -> IO ()
+run (Config w p) = do
+    f <- readFile p
+    putStrLn $ unlines $ wrapLine w =<< lines f
+
+wrapLine :: Int -> String -> [String]
+wrapLine w s = foldl (splitWordsAt w) [] $ words s
 
 splitWordsAt :: Int -> [String] -> String -> [String]
 splitWordsAt _ [] x = [x]
-splitWordsAt n xs x
-  | lineIsTooLong n (appendToLast xs x) = xs ++ [x]
+splitWordsAt w xs x
+  | length (appendToLast xs x) > w = xs ++ [x]
```

Comment:
not a big deal but isn't appending to the end not great for runtime optimization? Might be worth considering appending to the front and reversing at the end if you start to see performance hits.